cosmpy/aerial/client/__init__.py

```diff
@@ -0,0 +1,281 @@
+from cosmpy.crypto.address import Address
+from cosmpy.protos.cosmos.auth.v1beta1.auth_pb2 import BaseAccount
+from cosmpy.protos.cosmos.auth.v1beta1.query_pb2 import QueryAccountRequest
+from cosmpy.protos.cosmos.auth.v1beta1.query_pb2_grpc import QueryStub as AuthGrpcClient
+from cosmpy.protos.cosmos.bank.v1beta1.query_pb2 import QueryBalanceRequest
+from cosmpy.protos.cosmos.bank.v1beta1.query_pb2_grpc import QueryStub as BankGrpcClient
+from cosmpy.protos.cosmos.staking.v1beta1.query_pb2_grpc import (
+    QueryStub as StakingGrpcClient,
+)
+from cosmpy.protos.cosmos.tx.v1beta1.service_pb2 import (
+    BroadcastMode,
+    BroadcastTxRequest,
+    GetTxRequest,
+)
+from cosmpy.protos.cosmos.tx.v1beta1.service_pb2_grpc import ServiceStub as TxGrpcClient
+from cosmpy.protos.cosmwasm.wasm.v1.query_pb2_grpc import (
+    QueryStub as CosmWasmGrpcClient,
+)
+from cosmpy.staking.rest_client import StakingRestClient
+from cosmpy.tx.rest_client import TxRestClient
+
+DEFAULT_QUERY_TIMEOUT_SECS = 15
+DEFAULT_QUERY_INTERVAL_SECS = 2
+
+
+@dataclass
+class Account:
+    address: Address
+    number: int
+    sequence: int
+
+
+class LedgerClient:
+    def __init__(self, cfg: NetworkConfig):
+        cfg.validate()
+
+        self._network_config = cfg
+
+        parsed_url = parse_url(cfg.url)
+
+        if parsed_url.protocol == Protocol.GRPC:
+            if parsed_url.secure:
+                with open(certifi.where(), "rb") as f:
+                    trusted_certs = f.read()
+                credentials = grpc.ssl_channel_credentials(
+                    root_certificates=trusted_certs
+                )
+                grpc_client = grpc.secure_channel(parsed_url.host_and_port, credentials)
+            else:
+                grpc_client = grpc.insecure_channel(parsed_url.host_and_port)
+
+            self.wasm = CosmWasmGrpcClient(grpc_client)
+            self.auth = AuthGrpcClient(grpc_client)
+            self.txs = TxGrpcClient(grpc_client)
+            self.bank = BankGrpcClient(grpc_client)
+            self.staking = StakingGrpcClient(grpc_client)
+        else:
+            rest_client = RestClient(parsed_url.rest_url)
+
+            self.wasm = CosmWasmRestClient(rest_client)  # type: ignore
+            self.auth = AuthRestClient(rest_client)  # type: ignore
+            self.txs = TxRestClient(rest_client)  # type: ignore
+            self.bank = BankRestClient(rest_client)  # type: ignore
+            self.staking = StakingRestClient(rest_client)  # type: ignore
+
+    @property
+    def network_config(self) -> NetworkConfig:
+        return self._network_config
+
+    def query_account(self, address: Address) -> Account:
+        request = QueryAccountRequest(address=str(address))
+        response = self.auth.Account(request)
+
+        account = BaseAccount()
+        if not response.account.Is(BaseAccount.DESCRIPTOR):
+            raise RuntimeError("Unexpected account type returned from query")
```

Comment:
I think it would be nice to know what was returned

Reply:
Sure - tracked by #106